### PR TITLE
feature: ZENKO-734 lifecycle object processor for transitions

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Update the lifecycle object processor to support transition policies.

The object processor is a client of the data mover: when a data copy
is completed, it receives the results in the
lifecycle-object-tasks-topic, and updates the location in metadata
according to the new location sent by the data mover, on success, and
garbage-collects the old location.

### CODE REVIEW

**Review by bennettbuchanan — APPROVED** [review]

File: extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    _checkDate(entry, log, done) {
+        const bucket = entry.getAttribute('target.bucket');
+        const key = entry.getAttribute('target.key');
+        const lastModified = entry.getAttribute('details.lastModified');
+
+        if (lastModified) {
+            const reqParams = {
+                Bucket: bucket,
+                Key: key,
+                IfUnmodifiedSince: lastModified,
```
Comment (philipyoo) [inline extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js:29]: 👍

**Review by philipyoo — APPROVED** [review]

File: extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js (hunk 1 above)
Comment (jonathan-gramain) [reply extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js:29]: That was part of the original code in `LifecycleObjectTask.js`, only renamed the file :wink:

Comment (philipyoo) [reply extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js:29]: ah nice, I actually didn't know about this field and thought it was very useful

**Comment by jonathan-gramain** [comment]
@bert-e approve

**Pull request merged**